### DISCUSSION TITLE
Forecasts functionality

### DISCUSSION THREAD

**Discussion by mciantar** [firefly-iii/firefly-iii#8811] [General]
I have read and seen the position of the developer on Forecast. I come from a word of MS Money that was a really good tool. Looking at alternatives, I stumbled across many option, but they all lack something. Maybe GNUCash was the closest option, but in this day and age, I wanted to move past a desktop application and a web based approach seemed more suitable. 

Deploying Firefly-III was a breeze, and the Data-Importer is Excellent. I do understand the developer's view that budgeting the now is important, but once you have control on your spending, creating forecasts the understand better your investment growth and potential major life spendings is important and doing it right in your budgeting tool makes sense, as you already have all the Data.

The discussion to introduce forecast functionality was closed 2024.

> Has anything changed since then?
> If not, I have seen other users building "work-arounds" for this needed functionality. Are there any public projects that extend firefly that do this which someone can refer me to?

Thanks!

**Comment by inoculator** [comment]
Hi,
I know, this question is quite old, and the topic is well documented.
Maybe we can elaborate in this.
I am new to FF3 and have just started a self hosted server. Same experience to me as you describe for yours.
Now I am trying to create a forecasting process using the API.
My goal is to do it in a very simple approach by reading an account and its transactions, reading the recurrences and then process it into the future. No big financial speculations, just straight forward the transactions.
As an output I am targeting some sort of simple CSV that can then be processed in any program (LibreCalc/Excel/..etc).
My showstopper for this moment is the fact, that I am unable to interpret the values in a recurrence. So I am not able to process this items in any kind of forecasting.
Have you gained any progress on your goal for forecasting?

**Comment by Alex-AMC** [comment]
I am going to second this. There is no need to have fancy models for speculation. Just giving the user the ability to say, "My monthly income is X, my Monthly Expenditure is Y, with the leftover going to savings, what will the balance be in z months?" 
Then you can start adding things like "Now assume a 3% increase on expenditure every April 1st and increase salary by 2% every Jan 1st. What does the savings balance look like after J years? " 

I was half tempted to try and put something together but I've not looked at the API for this yet so I have no idea how easy it would be to add a custom tab to the dashboard

**Comment by inoculator** [comment]
After hearing of the FF API I launched a small project for my purpose.
Based on Powershell it reads simple account value and processes the automation parts, returning it as a grid view.
Very basic, but full fills my needs (see screenshot from a testrun)
<img width="1447" height="352" alt="image" src="https://github.com/user-attachments/assets/ffb97adf-1bf4-475c-8c18-f937c0dcd8f6" />
If You are interested, I can publish it as a GH Repo.

**Comment by extioStorm** [comment]
i want the budgeting software to allow me to forecast the future amount of flexable spending money after taking away constant recurring bills.  so i could put mortgage and power bill and days they will come out, and the graph shows a predicted view of them coming out.  then it tells me at a given point how much money i have to spend on fluctuating bills after that.  it would also warn me if i have 1200 dollars in account, but 1300 will come out before long, and only 600 will go back in.  therefore i know i absolutely must fight to keep over 500 to stay from missing a bill.

mind you i do this on my bill account, but i have a seperate spending account where i need more budgeting like the software actually does.

it helps people on the edge who already cut every expense but can barely keep up with life and forcibly living paycheck to paycheck despite divine wisdom in spending habits.
